src/components/vm/vmDetailsPage.jsx

```diff
@@ -64,6 +62,9 @@ export const VmDetailsPage = ({
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
+    // We want to reset the ConsoleState when a machine starts or shuts down.
+    const consoleState = useStateObject(() => new ConsoleState(), [vm.state]);
```

Comment:
This syntax seems weird. Not a blocker but would love to figure out a better way here.

Reply:
I really dislike Redux in general and feel it is way to complex for most use cases (machines being one of them). But yeah just a grievance of mine as both React and these related projects like Redux and React Router are just kinda meh today compared to alternatives. Would rather go with [Zustand](https://zustand.docs.pmnd.rs/getting-started/introduction) or [Jotai](https://jotai.org/) or something else that is smaller and more lightweight.

Not pushing for a refactor just sharing my thoughts once more :D